lib/utils/docker.ts

```diff
@@ -72,7 +72,9 @@ export const dockerConnectionCliFlags: flags.Input<DockerConnectionCliFlags> = {
 
 export const dockerCliFlags: flags.Input<DockerCliFlags> = {
 	tag: flags.string({
-		description: 'The alias to the generated image',
+		description: `\
+Tag any built images with this value. This is the 'tag' portion in
+'projectName_serviceName:tag'. The default is 'latest'.`,
```

Comment:
@dfunckt, re issue #825 that you created 3 years ago, and
@pipex re issue #2327 that you created a few hours ago,
you had the expectation that `--tag` could be used to specify the full / exact name of an image, e.g. `--tag my_img:my_tag`. The complication is that `balena build` or `balena deploy` may create several images, one for each service, and of course they cannot all have the same name. What this PR is currently implementing is using `--tag` only for the `:tag` part of the image name. The thinking is that it might be useful in the scenario where all service images should be tagged with, say, a common version for the app/fleet release. Example:

`$ balena build . --projectName dronecontrol --tag v2.2.4 ...`

Images created:
`dronecontrol_service1:v2.2.4`
`dronecontrol_service2:v2.2.4`
`dronecontrol_service3:v2.2.4`

What do you think? The `--tag` flag was being completely ignored for at least 3 years, so there are no worries about breaking compatibility at this point. I understand it might have worked a very a long time ago, before apps/fleets could have multiple services/containers, and `--tag` would then name the only image built.

Another option is to simply delete the `--tag` flag.

Reply:
@pdcastro I agree that `--tag` should merely deal with the tag part of the image name. To address Felipe's use case, one would then have to just do:

```
$ balena build --projectName myfleet --tag v1.1.0
```

Which would produce something like:

```
myfleet_api:v1.1.0
myfleet_sensor:v1.1.0
myfleet_dashboard:v1.1.0
```

Reply:
While we're at it, if it's not much of a complication I think it would be useful if `--tag` could be passed several times, to make several tags for images the same build, eg. both versioned *and* `latest`.

Reply:
This looks great @pdcastro, nice work. I think this definitely addresses the use case I saw on support this week.

My only worry would be with user confusion regarding diverging behavior between `balena build --tag` and `docker build --tag` (or `balena build` on balena engine). Even though that is addressed in the docs, I'm sure we'll get some confused users on support.

Also, what is the default value for `projectName` I assume is the folder name? We might want to clarify that somewhere if we haven't yet.

Reply:
I see your point. Technically, the direct equivalent for the CLI's `build` command is no longer Docker CLI (since balena is no longer single-container) but Docker Compose, which has no `--tag` attribute. Not sure how apparent is that to users though.

Reply:
> Also, what is the default value for projectName I assume is the folder name? We might want to clarify that somewhere if we haven't yet.

The help output for the `--projectName` flag reads:

Before this PR:
```
-n, --projectName <projectName>  Specify an alternate project name; default is the directory name
```
After this PR (as it stands):
```
-n, --projectName <projectName>  Prefix for the name of built images. This is the 'projectName' portion in
                                 'projectName_serviceName:tag'. The default is the directory name.

```

Reply:
> My only worry would be with user confusion regarding diverging behavior between balena build --tag and docker build --tag (or balena build on balena engine).

Even if we wanted to, the `docker build --tag` semantics cannot be used for the balena CLI, because `balena build` produces multiple images while `docker build` produces a single image. Like Akis pointed out, `balena build`'s counterpart is be closer to `docker-compose build` than `docker build`. We would have to come up with a different solution / semantic, or remove `--tag` altogether.

Reply:
> While we're at it, if it's not much of a complication I think it would be useful if `--tag` could be passed several times, to make several tags for images the same build, eg. both versioned _and_ latest.

I gather that the clean/proper way of doing this would be to change the interface between `balena-multibuild` and `balena-cli` and `balena-builder`, specifically changing the following `task.tag` attribute from `string` to `string[]` on this line: https://github.com/balena-io-modules/balena-multibuild/blob/v4.11.1/lib/build-task.ts#L55-L59

This is because it is `balena-multibuild` that sets the `t` attribute in the docker engine API request (which apparently does allow specifying multiple `t` attributes). It would then become 3 PRs plus testing, touching the builder. It could be done, yes, but I will play the _"is it a priority?"_ get-out-of-jail-free card, and leave it for the future. :-)

Reply:
> change the interface between balena-multibuild and balena-cli and balena-builder

Ah, good call -- I didn't realise this was the case, thanks for checking 👍

I was looking to reap low-hanging fruit, but it doesn't seem to be the case unfortunately.